build.rs

```diff
@@ -0,0 +1,53 @@
+// Copyright 2021 TiKV Project Authors. Licensed under Apache-2.0.
+
+use std::env;
+use std::path::Path;
+
+fn main() {
+    locate_coprocessor_plugins(&["example-plugin"]);
+}
+
+/// Locates compiled coprocessor plugins and makes their paths available via environment variables.
+///
+/// This is a bit hacky, but can be removed as soon as RFC
+/// <https://rust-lang.github.io/rfcs/3028-cargo-binary-dependencies.html> is available.
+///
+/// Why do we need this?
+/// This is because we want to use some coprocessor tests during tests. We can't just add them as a
+/// `dev-dependency`, because of how the CI works. Thus, we want to include them in our tests with
+/// `include_bytes!()`. However, we don't really know where the artifacts are located, so we need to
+/// find them with this function and expose the locations via environment variables.
+///
+/// Coprocessor plugins need to be added as a `build-dependency` in `Cargo.toml` for TiKV.
+fn locate_coprocessor_plugins(plugin_names: &[&str]) {
+    let out_dir = env::var_os("OUT_DIR").unwrap();
+    let build_dir = Path::new(&out_dir)
```

Comment:
If the behavior that `{OUT_DIR}/../../../`contains built libraries documented in cargo documentation? If not, I suggest just report a warning instead of panic with `unwrap` here.

Reply:
hm good point! Unfortunately, I think it might not work when `--target-dir` or `--out-dir` is specified :( 
see https://doc.rust-lang.org/cargo/commands/cargo-build.html#output-options
The only thing I found and that should always work is to search in `LD_LIBRARY_PATH`, because Cargo will modify it accordingly (https://doc.rust-lang.org/cargo/reference/environment-variables.html#dynamic-library-paths).
So maybe we should go with `LD_LIBRARY_PATH` instead?